lightning/src/ln/outbound_payment.rs

```diff
@@ -1134,21 +1177,9 @@ impl OutboundPayments {
 							log_error!(logger, "Payment not yet sent");
 							return
 						},
-						PendingOutboundPayment::InvoiceReceived { payment_hash, retry_strategy, .. } => {
-							let total_amount = route_params.final_value_msat;
-							let recipient_onion = RecipientOnionFields {
-								payment_secret: None,
-								payment_metadata: None,
-								custom_tlvs: vec![],
-							};
-							let retry_strategy = Some(*retry_strategy);
-							let payment_params = Some(route_params.payment_params.clone());
-							let (retryable_payment, onion_session_privs) = self.create_pending_payment(
-								*payment_hash, recipient_onion.clone(), None, &route,
-								retry_strategy, payment_params, entropy_source, best_block_height
-							);
-							*payment.into_mut() = retryable_payment;
-							(total_amount, recipient_onion, None, onion_session_privs)
+						PendingOutboundPayment::InvoiceReceived { .. } => {
+							log_error!(logger, "Payment already initiating");
```

Comment:
We could though, I'm wondering also if we should make this a `log_trace`. Given we'll send multiple invoice requests, we'll likely see multiple invoices even though we'll only handle the first one. The others may hit this case since the lock is released after transition to this state while path finding but before the payment is made.

Reply:
Actually, I was mistaken. We should never hit this case just like `PendingOutboundPayment::AwaitingInvoice`. When we release the lock in `send_payment_for_bolt12_invoice`, the payment will have transitioned to `PendingOutboundPayment::AwaitingInvoice`. So attempting to handle another invoice with the same id will cause us to return `Bolt12PaymentError::DuplicateInvoice`.

Reply:
Done and did the same for `PendingOutboundPayment::AwaitingInvoice`.